feat: add ethereum rpc table component and example

yqrashawn opened the pull request:
## Description

- add rpc table component through `ethereum-rpc-table` package
- add example usage in `docs/en/user-guide/setup.mdx`, this is just an example, the table is not supposed to be there, **do not merge**
- package code at https://github.com/scroll-tech/ethereum-rpc-table

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205686187705236

yqrashawn committed:
feat: add ethereum rpc table component and example

- add rpc table component through `ethereum-rpc-table` package
- add example usage in `docs/en/user-guide/setup.mdx`
- package code at https://github.com/scroll-tech/ethereum-rpc-table

Signed-off-by: yqrashawn <namy.19@gmail.com>

yqrashawn commented on src/components/RPCTable/RPCTable.astro:
```diff
@@ -0,0 +1,85 @@
+    },
+    {
+      url: "https://rpc.ankr.com/scroll",
+    },
+    {
+      url: "https://scroll-mainnet.chainstacklabs.com",
+    },
+  ],
+  faucets: [],
+  nativeCurrency: {
+    name: "Ether",
+    symbol: "ETH",
+    decimals: 18,
+  },
+  infoURL: "https://scroll.io",
+  shortName: "scr",
+  chainId: 534352,
+  networkId: 534352,
+  explorers: [
+    {
+      name: "Scrollscan",
+      url: "https://scrollscan.com",
+      standard: "EIP3091",
+    },
+    {
+      name: "Blockscout",
+      url: "https://blockscout.scroll.io",
+      standard: "EIP3091",
+    },
+  ],
+  parent: {
+    type: "L2",
+    chain: "eip155-1",
+    bridges: [
+      {
+        url: "https://scroll.io/bridge",
+      },
+    ],
+  },
+}
```
example data from https://chainlist.org/_next/data/rZLRgM60FTmTVWO5jkIXw/chain/534352.json?chain=534352

dghelm replied:
This is good for now!

yqrashawn commented on src/components/RPCTable/RPCTable.astro:
```diff
@@ -0,0 +1,85 @@
+  },
+}
+
+copy = copy || {
+  "RPC Server Address": "RPC Server Address",
+  Height: "Height",
+  Latency: "Latency",
+  Score: "Score",
+  Privacy: "Privacy",
+  "Add Network": "Add Network",
+  Connect: "Connect",
+  Sorting: "Sorting",
+}
```
all copy in this component, for i18n

yqrashawn commented on src/content/docs/en/user-guide/setup.mdx:
```diff
@@ -31,6 +32,8 @@ The **Add to wallet** links may not be compatible with every wallet. If you are
 | Currency Symbol    | ETH                                                              | ETH                                                                          |
 | Block Explorer URL | [https://sepolia.scrollscan.com](https://sepolia.scrollscan.com) | [https://sepolia.etherscan.io](https://sepolia.etherscan.io)                 |
 
+<RPCTable />
```
Example usage

**please put the table in the appropriate place or delete this before merge**

dghelm commented:
Question for @zzq0826 and @Turupawn -- Do we want i18n strings to live close to components, or would we prefer everything for a single language to sit in its `translation.json` file?

zzq0826 commented:
> Question for @zzq0826 and @Turupawn -- Do we want i18n strings to live close to components, or would we prefer everything for a single language to sit in its `translation.json` file?

If possible, I prefer to configure it in the` translation.json` file.

yqrashawn commented:
It's still possible to have i18n strings in [lang]/translation.json with this component.
I've updated the code to relfect that. I also added en and zh translation to json file.

@dghelm @zzq0826

yqrashawn commented:
To add custom style, you can remove
`import "ethereum-rpc-table/lib/default.css"` and `import "./RPCTable.css"` at https://github.com/scroll-tech/scroll-documentation/blob/6d0db315664e9607cdef0b3587277c7b8562ff0a/src/components/RPCTable/RPCTable.astro#L3-L5

The package adds zero style if these two imports are removed

dghelm committed:
move to components list and tweak css

dghelm reviewed (approved):
LGTM -- adjusted css and moved to components preview page.

Might assess followup for mobile, but we'll likely just want to hide it and make a component that alternates display after different breakpoints.
